src/numberToHangul/numberToHangul.ts

```diff
@@ -0,0 +1,53 @@
+import { HANGUL_CARDINAL, HANGUL_DIGITS, HANGUL_NUMBERS } from '@/_internal/constants';
+
+export function numberToHangul(input: number, options?: { spacing?: boolean }): string {
+  if (input === 0) {
+    return '영';
+  }
+
+  const koreanParts: string[] = [];
+  let remainingDigits = input.toString();
+  let placeIndex = 0;
+
+  while (remainingDigits.length > 0) {
+    const currentPart = remainingDigits.slice(-4);
+
+    koreanParts.unshift(`${numberToKoreanUpToThousand(Number(currentPart))}${HANGUL_DIGITS[placeIndex]}`);
+
+    remainingDigits = remainingDigits.slice(0, -4);
+    placeIndex++;
+  }
+
+  if (options?.spacing) {
+    return koreanParts
+      .filter(part => part !== '')
+      .join(' ')
+      .trim();
+  }
+
+  return koreanParts.join('');
+}
+
+function numberToKoreanUpToThousand(num: number): string {
+  if (num < 0 || num > 9999) {
+    throw new Error('0 이상 9999 이하의 숫자만 입력 가능합니다.');
+  }
+
+  let result = '';
+  let digitIndex = 0;
+
+  while (num > 0) {
+    const currentDigit = num % 10;
+
+    if (currentDigit > 0) {
+      result = HANGUL_NUMBERS[currentDigit] + HANGUL_CARDINAL[digitIndex] + result;
+    }
+
+    num = Math.floor(num / 10);
+    digitIndex++;
+  }
+
+  result = result.replace(/일천/, '천').replace(/일백/, '백').replace(/일십/, '십');
+
+  return result || '';
```

Comment:
아래와 같이 배열에 기본 함수들을 사용해보는것은 어떨까요?
let으로 선언된 변수들과, while문을 눈으로 따라가며 읽으니 약간 가독성이 좋지 않다고 생각했어요!

```suggestion
  const koreanDigits = num
    .toString()
    .split('')
    .reverse()
    .map((digit, index) => (digit === '0' ? '' : HANGUL_NUMBERS[Number(digit)] + HANGUL_CARDINAL[index]))
    .reverse()
    .join('');

  return koreanDigits.replace(/일천/, '천').replace(/일백/, '백').replace(/일십/, '십') || '';

```

Reply:
<img width="779" alt="Screenshot 2024-10-10 at 11 38 35 PM" src="https://github.com/user-attachments/assets/99ae962b-6c49-4b56-8a87-fcd431100c4f">

<img width="1282" alt="Screenshot 2024-10-10 at 11 38 59 PM" src="https://github.com/user-attachments/assets/89a52802-92b9-49bf-bcca-7f5d09734e65">


마다할 이유가 없네요 LGTM 👍🏻👍🏻👍🏻👍🏻

간단한 성능 측정도 해보았는데 성능상의 이점도 챙길 수 있겠어요!